applications/RomApplication/tests/structural_dynamic_test_files/test_ROM.py

```diff
@@ -27,10 +27,11 @@ def test_Struct_Dynamic_ROM_2D(self):
                 UP=0
                 DOWN=0
                 for j in range((np.shape(ObtainedOutput)[0])):
-                    UP += (NodalArea[j]*(    (ExpectedOutput[j,i] - ObtainedOutput[j,i]   )**2)  )
-                    DOWN +=  NodalArea[j]
-                L2 = np.sqrt(UP/DOWN)
-                self.assertLess(L2, 1.0e-4)
+                    if ExpectedOutput[j,i] != 0:
+                        UP += (NodalArea[j]*(    (1  - ((ObtainedOutput[j,i]) / (ExpectedOutput[j,i]) )   )**2)  )
+                        DOWN +=  NodalArea[j]
+                L2 = (np.sqrt(UP/DOWN)) *100
+                self.assertLess(L2, 1e-7) #percent
```

Comment:
this is still large...doesn't it work for 1e-12 or less?

Reply:
It does not, I think it is due to the dimensions used in this test. That is the displacements here are of the order of 1e-6, so the accuracy of the representation cannot be larger than this...

Reply:
ok, what you could do is to use the relative displacement (i mean to divide by the norm of the reference). that should be of the order 1e-12 or less 

if it doesn't work i am pretty convinced you have a problem